lsp-completion.el

```diff
@@ -0,0 +1,542 @@
+;;; lsp-completion.el --- LSP completion -*- lexical-binding: t; -*-
+;;
+;; Copyright (C) 2020 emacs-lsp maintainers
+;;
+;; This program is free software; you can redistribute it and/or modify
+;; it under the terms of the GNU General Public License as published by
+;; the Free Software Foundation, either version 3 of the License, or
+;; (at your option) any later version.
+
+;; This program is distributed in the hope that it will be useful,
+;; but WITHOUT ANY WARRANTY; without even the implied warranty of
+;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+;; GNU General Public License for more details.
+
+;; You should have received a copy of the GNU General Public License
+;; along with this program.  If not, see <https://www.gnu.org/licenses/>.
+;;
+;;; Commentary:
+;;
+;;  LSP completion
+;;
+;;; Code:
+
+(require 'lsp-mode)
+
+(define-obsolete-variable-alias 'lsp-prefer-capf
+  'lsp-completion-prefer-capf  "lsp-mode 7.0.1")
+
+(defcustom lsp-completion-prefer-capf nil
```

Comment:
can you rename that to lsp-completion-provider (or something like that)? We did similar transformation for lsp-prefer-flymake - it was renamed to lsp-diagnostics-package with :auto/:flymake/:flycheck/:none (t and nil are kept for backward compatibility). as values.

Reply:
Good idea